### ISSUE TITLE
NeoVim doesn't clear/recover tmux pane title upon exit

### ISSUE THREAD

**Issue by oblitum** [neovim/neovim#9572]
As [per comment](https://github.com/neovim/neovim/issues/4063#issuecomment-459877761):

> @oblitum As explained in #9408 , I tested this in tmux 2.7. Title stacking works. If it's not working, you need to figure out what's different about your environment. Provide steps to reproduce the issue with `nvim -u NORC`, then open a ticket.

---

- `nvim --version`:  0.3.4
- Vim (version: 8.1.751) behaves differently? **Yes**
- Operating system/version: ArchLinux
- Terminal name/version: termite v14
- `$TERM`: tmux-256color
- tmux version: 2.8

### Steps to reproduce using `nvim -u NORC`

In a fresh tmux pane,

```
tmux display-message -p 'title:<#{pane_title}>'
nvim -u NORC
:set title
:q
tmux display-message -p 'title:<#{pane_title}>'
```

### Actual behaviour

NeoVim changes tmux pane title but upon leaving it's not cleared in any form (recovering last title or simply clearing it up). Vim does clears it up upon exit.

### Expected behaviour

NeoVim should recover or clear the title, or provide some mechanism for doing so.

**Comment by oblitum** [comment]
Notice I'm on tmux 2.8, not 2.7.

**Comment by justinmk** [comment]
I can't imagine why tmux 2.8 would remove the title-stacking feature, so I don't think 2.7 vs 2.8 is relevant. Maybe 2.8 made it optional; if so, consult tmux's documentation.

Here I've tried again with tmux 2.7, and it works correctly:

```
$ tmux display-message -p 'title:<#{pane_title}>'
title:<gleipnir>
$ nvim -u NORC +'set title'
:!tmux display-message -p 'title:<\#{pane_title}>'
title:<[No Name] - NVIM>
:q
$ tmux display-message -p 'title:<#{pane_title}>'
title:<gleipnir>
```

Can you try the 0.4.0 `nvim.appimage` to see if it makes a difference: https://github.com/neovim/neovim/releases/tag/nightly

**Comment by oblitum** [comment]
@justinmk it gets recovered, so in my view it's behaving better than vim, that clears it.

**Comment by justinmk** [comment]
@oblitum 	If you want to clear it you could just set the `titleold` option.

So can this issue be closed?

**Comment by oblitum** [comment]
I mean, it gets recovered, just like in your session. I just dunno where or which commit  it got fixed for my side, since I'm on 0.3.4 and was supposed to have this fix already... I think it can be closed... though ArchLinux currently doesn't have a fixed version in their repos.

**Comment by justinmk** [comment]
https://github.com/neovim/neovim/commit/ccbcd390d42d33a15f15c29fab5d6076a6d3ac08 is not in a `v0.3.x` tag, only on `master`, so it did not get merged in time for `v0.3.4`.

**Comment by oblitum** [comment]
Ah OK weird, I thought the commit history from release page could be trusted,  the commit at the top is about this:

- https://github.com/neovim/neovim/compare/v0.3.4...master
